tasks/main.yml

```diff
@@ -0,0 +1,46 @@
+---
+- name: Calculate facts
+  set_fact:
+    build_base: "{{ (mumble_web.build_base | default(mumble_web_build_base)) | regex_replace('\\/$', '') }}"
+    deploy_path: "{{ (mumble_web.deploy_path | default(mumble_web_deploy_path)) | regex_replace('\\/$', '') }}"
+    version: "{{ mumble_web.version }}"
+    checksum: "{{ mumble_web.checksum }}"
+    reinstall: "{{ mumble_web.reinstall | default(false)}}"
+    config: "{{ mumble_web.config | default({}) }}"
+
+- name: Calculate facts for installation
+  set_fact:
+    archive_file: "{{ global_cache_dir }}/mumble-web-{{ version }}.tar.gz"
+    build_dir: "{{ build_base }}/mumble-web-{{ version }}"
+    deploy_base: "{{ deploy_path | dirname }}"
+
+- name: Create deployment parent directory
+  file:
+    path: "{{ deploy_base }}"
+    state: directory
+
+- name: Calculate facts for checking if install is neccesarry
+  set_fact:
+    deployed_version_file: "{{ deploy_base }}/.{{ deploy_path | basename }}_ansible_deployed_version"
+
+- name: Make sure deployed version file exists
+  file:
+    path: "{{deployed_version_file}}"
+    state: touch
+
+- name: Read deployed version file
+  shell: "cat {{deployed_version_file}}"
+  register: deployed_version_output
+
+- name: Get deployed version
+  set_fact:
+    deployed_version: "{{deployed_version_output.stdout | trim }}"
```

Comment:
I think this is only used once, I suggest to trim it there. Every task takes time to execute.

Reply:
Yes. I tried to keep line [41](https://github.com/stuvusIT/mumble-web/pull/1/files#diff-2444ad0870f91f17ca6c2a5e96b26823R41) clean and the logic readable.
But I will move it there if the extra task is actually a performance problem.